ldi-core/ldes-client/tree-node-supplier/src/main/java/ldes/client/treenodesupplier/MemberSupplier.java

```diff
@@ -28,6 +29,9 @@ public SuppliedMember get() {
 		try {
 			return executorService.submit(treeNodeProcessor::getMember).get();
 		} catch (ExecutionException e) {
+			if (e.getCause() instanceof EndOfLdesException endOfLdesException) {
+				throw endOfLdesException;
+			}
```

Comment:
Can this be refactored without `instanceof`? For example by first catching `EndOfLdesException`, then catching `ExecutionException` and then `InterruptedException`.